### PR TITLE
HAR-7831 - placeholders for empty fields

### PR DESCRIPTION
**Pull request by mattConnHarbour**
companion to webapp PR: https://github.com/Harbour-Enterprises/Harbour-WebApp/pull/4657

Should also solve:
- HAR-7833 https://linear.app/harbour/issue/HAR-7833/as-a-signer-i-am-unable-to-see-my-fields-on-the-pdf-document-on-the
- HAR-7835 https://linear.app/harbour/issue/HAR-7835/superdoc-previews-pdf-sign-date-annotation-missing

### CODE REVIEW

File: packages/superdoc/src/components/HtmlViewer/HtmlViewer.vue [hunk 1]
```diff
@@ -1,66 +1,63 @@
...
+    emit('ready', props.documentId);
+  } catch (error) {
+    emit('error', error);
+    console.error('Error loading document', error);
+  }
 }
 
-onMounted(() => {
-    loadDocument(props.fileSource);
+onMounted(async () => {
```
Comment (harbournick) [inline packages/superdoc/src/components/HtmlViewer/HtmlViewer.vue]: why do we need async here if we aren't awaiting anything?

File: packages/superdoc/src/components/HtmlViewer/HtmlViewer.vue [hunk 2]
```diff
@@ -1,66 +1,63 @@
 <script setup>
-import { defineProps, onMounted, ref } from 'vue';
+import { defineProps, defineEmits, onMounted, ref } from 'vue';
```
Comment (harbournick) [inline packages/superdoc/src/components/HtmlViewer/HtmlViewer.vue]: You don't need to explicitly import defineProps or defineEmits - in fact they'll generate a warning. Let's remove them.

File: packages/superdoc/src/Superdoc.vue [hunk 3]
```diff
@@ -434,7 +434,10 @@ const handlePdfClick = (e) => {
             v-if="doc.type === PDF"
             :document-data="doc"
             @selection-change="handleSelectionChange"
-            @ready="handlePdfReady" 
+            @selection-drag="handleSelectionDrag"
```
Comment (harbournick) [inline packages/superdoc/src/Superdoc.vue]: Did you mean to re-add these emits here? I removed them yesterday. If not, can you remove? If yes, we should figure out why - the selection is now handled in Superdoc.vue now in the viewer.

**Review by harbournick — CHANGES_REQUESTED** [review]
Couple minor tweaks

File: packages/superdoc/src/components/HtmlViewer/HtmlViewer.vue (hunk 1 above)
Comment (mattConnHarbour) [reply packages/superdoc/src/components/HtmlViewer/HtmlViewer.vue]: removing

**Review by harbournick — APPROVED** [review]
LGTM

**Pull request merged**